frappe/utils/print_format.py

```diff
@@ -19,7 +19,35 @@
 
 
 @frappe.whitelist()
-def download_multi_pdf(doctype, name, format=None, no_letterhead=False, letterhead=None, options=None):
+def download_multi_pdf(
+	doctype,
+	name,
+	format=None,
+	no_letterhead=False,
+	letterhead=None,
+	options=None,
+	task_id=None,
+):
+	"""
+	Calls _download_multi_pdf with the given parameters and returns the URL of the generated PDF
+
+	"""
+	frappe.enqueue(
+		_download_multi_pdf,
+		doctype=doctype,
+		name=name,
+		format=format,
+		no_letterhead=no_letterhead,
+		letterhead=letterhead,
+		options=options,
+		task_id=task_id,
+	)
```

Comment:
You will need to use long queue here to avoid blocking other jobs.

Reply:
Should I make it dynamic, short queue for upto maybe 20 documents otherwise long?

Reply:
Works.

Reply:
Works.